### PR TITLE
feat: disallow personal access tokens for oauth users.

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-5723
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@kjaskiewiczz could you give me an early review here? before I dive into tests. just approve the general direction/API.

File: model/user.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ type User struct {
 
 	// LoginTs is the timestamp of the last login for this user.
 	LoginTs *time.Time `json:"login_ts,omitempty" bson:"login_ts,omitempty"`
+
+	// IsOauth is the indicator that the user is using oauth mechanism to login
+	IsOauth bool `json:"is_oauth,omitempty" bson:"is_oauth,omitempty"`
```
Comment (mzedel) [inline model/user.go:81]: can we maybe make this a string and get the type of provider in there (<github|google|microsoft>) ? this way the UI can show some context to the user about where to change their login information...

Comment (merlin-northern) [reply model/user.go:81]: of course we can. (this is the main idea why I ask for "early reviews")

Comment (merlin-northern) [reply model/user.go:81]: all I have is the `login` field, so I will just add here what is in the login after the '@' will that work? parsing the arbitrary domains and deciding if it is google or something else is too much for me. (thats also why I added only boolean here, it was the cleanest solution)

Comment (merlin-northern) [reply model/user.go:81]: alternatively I can do `google` if login matches `.*@.*(gmail|googlemail).com*` and for the remaining two the same, while `unknown` for others, and `""` is not enabled.

Comment (mzedel) [reply model/user.go:81]: found it: https://github.com/mendersoftware/useradm-enterprise/blob/1.17.x/model/user.go#L78 <- that used to be accessible in the frontend... can we please have it back?

Comment (merlin-northern) [reply model/user.go:81]: yes but it is beyond the scope of this task. can you join todays BET meeting and present your case?

Comment (merlin-northern) [reply model/user.go:81]: we discussed. closing this one.

**Pull request closed without merging**